compiler/p/codegen/OMRTreeEvaluator.cpp

```diff
@@ -5795,12 +5795,17 @@ TR::Register *OMR::Power::TreeEvaluator::sbyteswapEvaluator(TR::Node *node, TR::
          {
          generateTrg1Src1Instruction(cg, TR::InstOpCode::brh, node, tgtRegister, srcRegister);
          }
+      else if (cg->comp()->target().cpu.isAtLeast(OMR_PROCESSOR_PPC_P8))
+         {
+         generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwimi, node, tgtRegister, srcRegister, 24, CONSTANT64(0x00000000ff));
+         generateTrg1Src1Imm2Instruction(cg, TR::InstOpCode::rlwimi, node, tgtRegister, srcRegister, 8,  CONSTANT64(0x000000ff00));
+         }
```

Comment:
Would it be possible to write this as an `rlwinm` and an `rldimi` to avoid having to maintain two separate implementations here? IIUC `rldimi` doesn't suffer the same penalty as `rlwimi` pre-P8.

Reply:
True.. it can be one implementation for all P7 - P9. Will fix it.

Reply:
Fixed in bb0c249. Short byteswap may change further as modifications gives slower perf on P8.